### ISSUE TITLE
Bug: Export to Matplotlib while scientific notation is being used 

### ISSUE THREAD

**Issue by penkod** [pyqtgraph/pyqtgraph#1050]
Pyqgraph by default uses scientific notation, meaning that when values higher than 1000000 are present, all such values are reduced to e.g. "<reduced value> x1e+06" and the plot label is changed accordingly. However, if we then try to export the plot to matplotlib window, the original values are kept **AND** the (now wrong) label, which includes also the "(x1e+06)" part, making the plot state that the values are then **x 1e+12**.
![pyqtgraph_matplotlib_bug](https://user-images.githubusercontent.com/54578179/65887546-64621c00-e39e-11e9-8abf-48b9ab27d9a1.png)

**Comment by penkod** [comment]
The fix is quite easy. I've modified the exporters/Matplotlib.py file to keep the same label and values as in the pyqtgraph plot (at least for y-axis labels and values). I've included for values:

```
if "(x1e+06)" in ylabel:
    y = [a/1000000. for a in y]
```
and for view range: 

```
if "(x1e+06)" in ylabel:
    yr = [a/1000000. for a in yr] 
```
Now the label and values are both correct. 
![pyqtgraph_matplotlib_bug_fix](https://user-images.githubusercontent.com/54578179/65888721-48f81080-e3a0-11e9-975a-e50cdc0cea9b.png)

I could create a pull request for this bug but I believe I don't have the necessary writing permissions.

**Comment by ixjlyons** [comment]
Good catch, thanks for reporting. You don't need permissions to submit a pull request -- the workflow is fork the repository to your own account, commit to a new branch, push to your fork, then use the GitHub interface to create the pull request. Let us know if you need help. Also see https://help.github.com/en/articles/about-pull-requests

We do need a more general solution than above, however. The multiplier suffix appended to the label could be "x1e+09" or something else. I believe there are enough properties of `AxisItem` to determine the correct axis label and tick labels pretty easily, but I'd have to come back and take another look when I get more time. We also need to take into account the SI prefix on units when specified. That is currently broken as well.

**Comment by penkod** [comment]
Thank you for a quick response. 

I've added more general modifications to the code which checks for the
`self.item.axes['left']['item'].autoSIPrefixScale`
which is in my case **1e-06** (float), and sets those values and view range accordingly. 

I've included to first check
`if self.item.axes['left']['item'].autoSIPrefix == True`
though in all my cases, even if the values were not over 1ex06, the value was True.

I've set the code to check and modify values if needed for the 'bottom' axis too.

For my case, it works fine, also when having multiple curves. Same for plots that don't contain large values the plots remain fine. 

I've made a fork and a pull request following your instructions. I hope this my small contribution will help you to solve this issue.
